lib/ae_mdw/db/mutations/contract_call_mutation.ex

```diff
@@ -30,13 +43,47 @@ defmodule AeMdw.Db.ContractCallMutation do
 
   @spec mutate(t()) :: :ok
   def mutate(%__MODULE__{
+        contract_pk: contract_pk,
+        caller_pk: caller_pk,
         create_txi: create_txi,
         txi: txi,
         fun_arg_res: fun_arg_res,
         call_rec: call_rec
       }) do
     DBContract.call_write(create_txi, txi, fun_arg_res)
     DBContract.logs_write(create_txi, txi, call_rec)
+
+    with true <- Contract.is_aex9?(contract_pk),
+         {:ok, method_name, method_args} <-
+           Contract.extract_non_stateful_aex9_function(fun_arg_res) do
+      update_aex9_presence(
+        contract_pk,
+        caller_pk,
+        txi,
+        method_name,
+        method_args
+      )
+    end
+
+    :ok
+  end
+
+  #
+  # Private functions
+  #
+  defp update_aex9_presence(contract_pk, caller_pk, txi, method_name, method_args) do
+    account_pk =
+      if method_name in ["burn", "swap"] do
```

Comment:
both require presence before being used, so can be ignored for further presence finding

Reply:
The mdw is tracking the presence in two ways: 
- on creation (`create_txi`) 
- and per change basis (`call_txi`)

This would be for the `call_txi`.

Reply:
what gets "changed" for burn and swap?

Reply:
I understand your concern and you know better than me that the balance of the caller is burned partially or completely. 

The thing is that even though the presence is already mapped, the model is `{account_pk, create_or_call_txi, contract_pk}` and the txi of the last change (last `call_txi`) is shown in the [/aex9/balances/account/:account](https://mainnet.aeternity.art/mdw/aex9/balances/account/ak_rLoCtHE3NK9dKyCNonJFYWkEEfeAsDUWa887GsCKqV1rhSuT6) 

The block_hash, tx_hash, tx_type and height are also derived from the last `call_txi`.

Reply:
One improvement that could be done in the model would be change it to:
`{account_pk, contract_pk, create_txi, call_txi}` because only the last `call_txi` matters. 

It's retrieved only by `aex9_search_contract` that is called by `account_balances_reply` that filters by:
```
        call_txi = List.last(txi_list)
        {amount, create_txi, call_txi, contract_pk}
```
PS: Before this enhancement, with the current model, it could check if already exists before writing a new but it might degrade because it needs to iterate throughout all call_txis until find one that matches the contract_pk.
Opening an enhancement issue for this ...

Reply:
Please find it here #459

Reply:
okay, looks good then :)